plugin/settings/checkbox.php

```diff
@@ -13,28 +13,52 @@
  * This will become unnecessary when using the new AJAX/API/Form module, because
  * they use a smart JavaScript function to gather form fields data.
  */
-function render_setting_field_checkbox($config) {
+function render_setting_field_checkbox($config, $type='checkbox') {
 
   foreach ([
     'name',
     'value',
-    'label'
+    'label',
+    'description'
   ] as $key) {
     $$key = $config[$key];
   }
 
   $checked = $value==='true';
 
-  ?>
-  <label>
-    <input type="hidden" name="<?php echo $name; ?>" value="<?php
-      echo $checked ? 'true' : 'false';
-    ?>" autocomplete="off"><input type="checkbox"
-      value="true" autocomplete="off"
-      onclick="this.previousSibling.value=this.previousSibling.value==='true'?'false':'true'"
-      <?php echo $checked ? 'checked="checked"' : ''; ?>
-    />
-    <?php echo esc_html($label); ?>
-  </label>
-  <?php
+  if ($type === 'checkbox'): ?>
```

Comment:
```suggestion
  if (empty($type)): ?>
```

It's `checkbox` by default for backward compatibility.